cmd/tarianctl/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,215 @@
+	cmd := &uninstallCmd{
+		globalFlags: globalFlags,
+		logger:      log.GetLogger(),
+	}
+
+	uninstallCmd := &cobra.Command{
+		Use:   "uninstall",
+		Short: "Uninstall Tarian from a Kubernetes cluster",
+		PreRun: func(_ *cobra.Command, _ []string) {
+			// cmd.logger.Warn("WARNING: This command is in alpha version and may have some bugs and breaking changes in future releases.")
+			yellow := "\033[33m"
+			reset := "\033[0m"
+			text := "WARNING: This command is in alpha version and may have some bugs and breaking changes in future releases."
+			fmt.Printf("%s%s%s\n", yellow, text, reset)
+		},
+		RunE: cmd.run,
+	}
+	uninstallCmd.Flags().StringVar(&cmd.namespace, "namespace", "tarian-system", "Namespace to uninstall Tarian from")
+	uninstallCmd.Flags().BoolVar(&cmd.onlyAgents, "only-agents", false, "Set to uninstall Tarian Cluster Agent and Node Agent only")
+	uninstallCmd.Flags().StringVar(&cmd.kubeContext, "kube-context", "", "Kubernetes context to use")
+	uninstallCmd.Flags().StringVar(&cmd.kubeconfig, "kubeconfig", "", "Kubernetes configuration file")
+
+	return uninstallCmd
+}
+
+func (c *uninstallCmd) run(cmd *cobra.Command, args []string) error {
+	var err error
+	// set kubeconfig
+	if c.kubeconfig == "" {
+		if kubeconfigEnv := os.Getenv("KUBECONFIG"); kubeconfigEnv != "" {
+			c.logger.Debugf("using kubeconfig from environment variable, KUBECONFIG=%s", kubeconfigEnv)
+			c.kubeconfig = kubeconfigEnv
+		} else {
+			c.logger.Debug("using kubeconfig from default kubeconfig path ~/.kube/config")
+			home := homedir.HomeDir()
+			c.kubeconfig = filepath.Join(home, ".kube", "config")
+			_, err := os.Stat(c.kubeconfig)
+			if err != nil {
+				c.logger.Debug("could not find kubeconfig in default path ~/.kube/config")
+				c.logger.Error("Please set the --kubeconfig flag or the KUBECONFIG environment variable")
+				return fmt.Errorf("uninstall: could not find kubeconfig")
+			}
+		}
+	}
+
+	if c.kubeContext != "" {
+		c.logger.Debug("using kubeconfig context: ", c.kubeContext)
+	}
+	c.logger.Debug("Using Namespace: ", c.namespace)
+	c.logger.Infof("Uninstalling Tarian in namespace '%v'...", c.namespace)
+
+	// Create Helm and Kubernetes client instances
+	if c.helmClient == nil {
+		c.helmClient, err = helm.NewHelmClient(c.logger, c.kubeconfig, c.kubeContext)
+		if err != nil {
+			return fmt.Errorf("uninstall: failed to create helm client: %w", err)
+		}
+	}
+
+	if c.kubeClient == nil {
+		c.kubeClient, err = kubeclient.NewKubeClient(c.logger, c.kubeconfig, c.kubeContext)
+		if err != nil {
+			return fmt.Errorf("uninstall: failed to get kubeclient: %w", err)
+		}
+	}
+
+	var g errgroup.Group
+
+	for _, f := range []func() error{
+		c.uninstallAgents,
+		c.uninstallServer,
+		c.uninstallDgraph,
+		c.uninstallNats,
+	} {
+		f := f
+		g.Go(func() error {
+			return f()
+		})
+	}
+	err = g.Wait()
+	if err != nil {
+		return fmt.Errorf("uninstall: failed to uninstall: %w", err)
+	}
+
+	if c.onlyAgents {
```

Comment:
I don't think uninstalling onlyAgents would work. Can you check it once and let me know the results?

Reply:
Hi @pratikjagrut , I have added unit test cases for that, May be need some more information to investigate